components/dashboard/src/service/public-api.ts

```diff
@@ -275,12 +275,24 @@ export function stream<Response>(
     (async () => {
         while (!abort.signal.aborted) {
             try {
+                let dataReceived = false;
```

Comment:
@filiptronicek I created https://github.com/gitpod-io/gitpod/pull/20176/files to use a class `Timeout` I needed for a similar problem in the frontend.